dspace-api/src/main/java/org/dspace/content/EntityServiceImpl.java

```diff
@@ -129,11 +168,18 @@ public EntityType getType(Context context, Entity entity) throws SQLException {
     }
 
     @Override
-    public List<RelationshipType> getRelationshipTypesByTypeName(Context context, String label) throws SQLException {
+    public List<RelationshipType> getRelationshipTypesByTypeName(Context context, String type) throws SQLException {
+
+        return getRelationshipTypesByTypeName(context, type, -1, -1);
+    }
+
+    @Override
+    public List<RelationshipType> getRelationshipTypesByTypeName(Context context, String type,
+                                                                 Integer limit, Integer offset) throws SQLException {
         List<RelationshipType> listToReturn = new LinkedList<>();
-        for (RelationshipType relationshipType : relationshipTypeService.findAll(context)) {
-            if (StringUtils.equals(relationshipType.getLeftwardType(),label) ||
-                StringUtils.equals(relationshipType.getRightwardType(),label)) {
+        for (RelationshipType relationshipType : relationshipTypeService.findAll(context, limit, offset)) {
+            if (StringUtils.equals(relationshipType.getLeftwardType(),type) ||
+                StringUtils.equals(relationshipType.getRightwardType(),type)) {
                 listToReturn.add(relationshipType);
```

Comment:
This produces result pages with 0 to `limit` entries. This may be unexpected. The filtering should be done before the paging.

Reply:
@AlexanderS I must of missed these when I took over the ticket. Thank you for pointing them out. 

I've addressed all the methods you've pointed out in your comment, amended related test, and touched up docs and method signatures in places that were still using `label` instead of `TypeName`.

[getRelationsByLabel](https://github.com/DSpace/DSpace/pull/2535/commits/cd531d81f025a89a7c2b17a9fbc789551c97d102#diff-b02101fc3c84839598e3272b448ac906R92)

[getAllRelationshipTypes](https://github.com/DSpace/DSpace/pull/2535/commits/cd531d81f025a89a7c2b17a9fbc789551c97d102#diff-b02101fc3c84839598e3272b448ac906R104)

[getLeftRelationshipTypes](https://github.com/DSpace/DSpace/pull/2535/commits/cd531d81f025a89a7c2b17a9fbc789551c97d102#diff-b02101fc3c84839598e3272b448ac906R110)

[getRightRelationshipTypes](https://github.com/DSpace/DSpace/pull/2535/commits/cd531d81f025a89a7c2b17a9fbc789551c97d102#diff-b02101fc3c84839598e3272b448ac906R122)

[getRelationshipTypesByTypeName](https://github.com/DSpace/DSpace/pull/2535/commits/cd531d81f025a89a7c2b17a9fbc789551c97d102#diff-b02101fc3c84839598e3272b448ac906R140)